src/hooks/useFullKeyboardListeners.js

```diff
@@ -0,0 +1,85 @@
+import { useMemo, useLayoutEffect, useCallback } from "react";
+import useKeyEvent from "./useKeyEvent";
+
+export const NAV_DIRECTIONS = {
+  UP: "up",
+  DOWN: "down",
+  LEFT: "left",
+  RIGHT: "right"
+};
+
+const ARROW_DOWN_KEYS = ["ArrowDown"];
+const ARROW_UP_KEYS = ["ArrowUp"];
+const ARROW_RIGHT_KEYS = ["ArrowRight"];
+const ARROW_LEFT_KEYS = ["ArrowLeft"];
+const SELECTION_KEYS = ["Enter", " "];
+const ESCAPE_KEYS = ["Escape"];
+
+const NOOP = () => {};
+
+export default function useFullKeyboardListeners({
```

Comment:
I saw that we had various places that add similar key listeners, but some had slightly different key-handling. For example, treating only "Enter" as selection key, and not "Space". 
If you're ok with it, I can replace some of the existing hooks to use this hook, to reduce code duplication and having a more unified behavior.